### DISCUSSION TITLE
Gigabyte X570 Aorus Elite

### DISCUSSION THREAD

**Discussion by CptSchwifty** [terminatorul/NvStrapsReBar#5] [General]
I'm so sorry, but i cant add the NvStrapsReBar.FFS to the Bios file (f38h, rev1).. Are these locked?
No matter what i try, i cant seem to flash it. Its an APTIO V Bios and even the modded efiflash binaries keep telling me they couldnt load the modified bios.. 
Has anybody found a solution to modify aptio5 bioses to use the ffs file ?

**Comment by dakisback2** [comment]
https://winraid.level1techs.com/t/guide-how-to-flash-a-modded-ami-uefi-bios/30627


See section for Gigabyte

**Comment by terminatorul** [comment]
Also adding this here for reference (same as above answer from @dakisback2), worked on X470 and X570:

I have a Gigabyte X470, and I followed the instructions here:
https://winraid.level1techs.com/t/flashing-gigabyte-while-avoiding-invalid-bios-image/31185

The original article is hard to read, but what you have to do is simple:

- open your final UEFI image, that you want to flash onto the motherboard, with a hex editor. I used gvim with :%!xxd command.
- find the ASCII string "$BDR" and note the resulting offset
- add 63h to that offset, and read the byte at the resulting offset. In my case it was 3, but other values are possible according to the article
- check that bit 0 in that value is set (check the byte value is an odd number)
- change the byte to reset bit 0 (subtract 1 from the original value)
- save the file (with gvim, also execute :%!xxd -r before saving)
- compare the original and the modified files, with a command like the Windows `FC /B` in a `cmd` console, and verfy that only a single byte was changed, and you recognize the old and new values of that byte
- reboot and enter UEFI setup
- find Q-Flash option in UEFI setup, usually on the main screen, and use it to flash the motherboard. Trying to use some other tools for flashing will not work....

Usually Q-Flash can read files from your Windows hard drive directly (to browse for the UEFI image file), but make sure the file system does not use any features like encryption or compression. Using a short path to the image file is best. Otherwise, if it does not work, use an USB flash drive instead.
